### DISCUSSION TITLE
How to catch 429 (Too Many Requests)

### DISCUSSION THREAD

**Discussion by LIMPIX31** [kaimallea/node-imgur#237] [Q&A]
doesn't work
```ts
try {
      imgur.upload(pathToImage);
} catch (e) {
      console.log('Too many uploads at the same time')
}
```

**Comment by KenEucker** [comment]
@LIMPIX31 

I think we could come up with a way to gracefully handle HTTP 429 responses. This happens often, and in other scenarios than just too many uploads.

NOTE:
This package is now being maintained in another repository: https://github.com/KenEucker/imgur and the new version might have already addressed your concerns.

imgur@v2 will be released before the end of 2021, and with that will come a couple of breaking changes -> dropping support for filesystem uploads means you have to create your streams on your own and pass them to the API.

Please come help us maintain v2!
